### PR TITLE
600 acc registration with delegation

### PR DESCRIPTION
**Pull request by Katkevich**
It is more convenient to review this PR commit-per-commit

### CODE REVIEW

**Comment by Katkevich** [comment]
It is more convenient to review this PR commit-per-commit

File: libraries/chain/include/scorum/chain/schema/account_objects.hpp [hunk 1]
```diff
@@ -406,6 +417,19 @@ typedef shared_multi_index_container<scorumpower_delegation_object,
...
     scorumpower_delegation_index;
 
+struct by_delegatee;
+
+typedef shared_multi_index_container<reg_pool_sp_delegation_object,
+                                     indexed_by<ordered_unique<tag<by_id>,
+                                                               member<reg_pool_sp_delegation_object,
+                                                                      reg_pool_sp_delegation_id_type,
+                                                                      &reg_pool_sp_delegation_object::id>>,
+                                                hashed_non_unique<tag<by_delegatee>,
```
[the remark this answers is not in the document]
Comment (alexplevako) [reply libraries/chain/include/scorum/chain/schema/account_objects.hpp]: it should be `unique`

File: libraries/chain/include/scorum/chain/schema/dynamic_global_property_object.hpp [hunk 2]
```diff
@@ -61,6 +61,9 @@ class dynamic_global_property_object
     /// Current head block time
     time_point_sec time;
 
+    // Blockchain genesis time
+    time_point_sec genesis_time;
```
[the remark this answers is not in the document]
Comment (alexplevako) [reply libraries/chain/include/scorum/chain/schema/dynamic_global_property_object.hpp]: lets move genesis_time to chain_property_object

File: tests/chain_tests/operation_tests.cpp [hunk 3]
```diff
@@ -3331,7 +3330,7 @@ BOOST_AUTO_TEST_CASE(decline_voting_rights_apply)
 
         db.get<comment_vote_object, by_comment_voter>(
             boost::make_tuple(db.obtain_service<dbs_comment>().get("alice", std::string("test")).id,
```
[the remark this answers is not in the document]
Comment (Katkevich) [reply tests/chain_tests/operation_tests.cpp]: I believe we need to refactor such code in separate task/branch. In this branch I've refactored the required minimum.

File: libraries/protocol/include/scorum/protocol/config.hpp [hunk 4]
```diff
@@ -224,6 +224,7 @@ namespace detail {
 
 #define SCORUM_CREATE_ACCOUNT_DELEGATION_RATIO     5
 #define SCORUM_CREATE_ACCOUNT_DELEGATION_TIME      fc::days(30)
+#define SCORUM_CREATE_ACCOUNT_REG_COMMITTEE_DELEGATION_MAX      10
```
Comment (alexplevako) [inline libraries/protocol/include/scorum/protocol/config.hpp]: this is should be 10000000000

**Pull request merged**